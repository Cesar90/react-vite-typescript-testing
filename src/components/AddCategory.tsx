import { useState, FC } from "react";

interface IProps {
  setCategories?: React.Dispatch<React.SetStateAction<string[]>>;
  onNewCategory: (category: string) => void;
}

export const AddCategory: FC<IProps> = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState<string>("");

  const onInputChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value);
  };

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCategories( (categories) => [inputValue, ...categories] )
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
      <input
        type="text"
        placeholder="Find Gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};
