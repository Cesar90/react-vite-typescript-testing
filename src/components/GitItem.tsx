import { FC } from 'react'

interface IProps{
    image: RequireOnly<Datum, 'id'|'url'|'title'>
}

type AnyCompProps = React.ComponentProps<typeof GitItem>


export const GitItem:FC<TImage> = ({id, url, title}) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        {title}
    </div>
  )
}
