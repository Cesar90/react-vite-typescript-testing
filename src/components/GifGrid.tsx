import { FC  } from "react"
import { GitItem } from "./GitItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

interface IProps{
    category: string
}

export const GifGrid:FC<IProps> = ({category}) => {
    const {images, isLoading} = useFetchGifs(category)

    return (
        <>
            <h3>
                {category}
            </h3>
            {
                isLoading && (<h2>Loading...</h2>)
            }
            <div className="card-grid">
                {
                    images.map((image) => {
                        return <GitItem key={image.id} {
                            ...image
                        } />
                    })
                }
            </div>
        </>
    )
}
