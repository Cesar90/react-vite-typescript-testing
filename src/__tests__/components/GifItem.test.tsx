import { render, screen } from "@testing-library/react";
import { GitItem } from "../../components/";

describe("Testing in <GitItem/ >", () => {
  const url = "https://goku-one.com";
  const title = "Goku";
  const id = "1";

  test("must match with snapshot", () => {
    const { container } = render(<GitItem url={url} title={title} id={id} />);
    expect(container).toMatchSnapshot();
  });

  test("must to show the title, url and id", () => {
    const { getByText } = render(<GitItem url={url} title={title} id={id} />);
    expect(getByText(title)).toBeTruthy();
  });

  test("must to show the imagen with the URL and ALT indicated", () => {
    render(<GitItem url={url} title={title} id={id} />);
    //screen.debug();
    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("src", url);
    expect(image).toHaveAttribute("alt", title);
  });
});
