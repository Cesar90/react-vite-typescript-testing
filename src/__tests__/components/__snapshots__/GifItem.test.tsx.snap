// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GitItem/ > must match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku-one.com"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
